lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -69,7 +72,11 @@
   private boolean renewFailedEvaluators = true;
   private ConfigurationModule driverConfigurationModule = null;
   private String inputFormatClass;
-  private String inputPath;
+  /**
+   * Partitions in the data. Can be thought as folders. Each folder or partition
+   * can contain several files.
+   */
+  private List<DataPartition> partitions = new ArrayList<>();
```

Comment:
Also, is it legal to have more than one copy of each partition? If not, we should make this a `Set`.